### PR TITLE
Feature/particles

### PR DESCRIPTION
**Pull request by lily-seabreeze**

### CODE REVIEW

File: demo/demo.py [hunk 1]
```diff
@@ -89,9 +97,44 @@
...
+)
+
+
+
+
+last_time = time.time()
 # Main program loop
 while not done:
+    new_time = time.time()
+    elapsed = new_time - last_time
```
Comment (lily-seabreeze) [inline demo/demo.py]: why not http://www.pygame.org/docs/ref/time.html#pygame.time.Clock.get_time

File: demo/demo.py [hunk 2]
```diff
@@ -137,11 +180,16 @@
...
     if potential_rect.collidelist(solid_blocks_on_players_index) != -1:
         print("colliding!")
+        x_speed = y_speed = 0
     else:
         y_coord = potential_y_coord
         x_coord = potential_x_coord
         camera.scroll_to(potential_rect)
         #camera.scroll_absolute(x_coord - 10, y_coord - 10)
+
+    source.x = x_coord + animated_sprite.image.get_size()[0] / 2
```
Comment (lily-seabreeze) [inline demo/demo.py]: `x_coord` will not work with demo once merged to `nextrelease`. instead, it'll have to be `animated_collisionsprite.rect.left` or whatever

File: demo/demo.py [hunk 3]
```diff
@@ -153,6 +201,8 @@
     # arbitrary layer. In the future the TMX will set this.
     layers[ANIMATED_SPRITE_Z_INDEX].blit(animated_sprite.image, (x_coord, y_coord))
 
+    fountain.draw_on(layers[ANIMATED_SPRITE_Z_INDEX])
```
Comment (lily-seabreeze) [inline demo/demo.py]: once merged to `nextrelease` `ANIMATED_SPRITE_Z_INDEX` is in `config`

File: demo/demo.py [hunk 4]
```diff
@@ -11,11 +11,16 @@
 """
  
 import pygame
+import time
+import os, sys
+sys.path.insert(0, os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+print(sys.path)
```
Comment (lily-seabreeze) [inline demo/demo.py]: yes, good

File: demo/demo.py [hunk 5]
```diff
@@ -37,6 +42,9 @@
 # The layer that the player's sprite will be rendered on 
 ANIMATED_SPRITE_Z_INDEX = 0
 
+# Number of frames per second
+FRAMES_PER_SECOND = 60
```
Comment (lily-seabreeze) [inline demo/demo.py]: once merged to `nextrelease` this should go in `config.py`

File: demo/demo.py [hunk 6]
```diff
@@ -89,9 +97,44 @@
...
+            (64, 63, 127, 0)
+        ],
+        #pygame.BLEND_RGB_MAX,
+    )
+)
+
+
+
+
+last_time = time.time()
```
Comment (lily-seabreeze) [inline demo/demo.py]: a lot of these time operations are atomated with pygame's `Clock`

File: sappho/particle.py [hunk 7]
```diff
@@ -0,0 +1,544 @@
...
+"""
+from __future__ import division
+import copy
+import math
+import random
+
+import pygame
+
+
+# The biggest number less than infinity-plus-one
```
Comment (lily-seabreeze) [inline sappho/particle.py]: hahaha <3

**Comment by lily-seabreeze** [comment]
This is amazing I love it! Very elegant, but simple! Very well documented! I've left some comments in the source, mostly what to beware of during the `nextrelease` merge conflict process.

**Comment by lily-seabreeze** [comment]
This is so amazing. I'm going to have to study this!

**Comment by lily-seabreeze** [comment]
Hey @kcsaff any notes on current status? Just curious. :smile:

**Comment by kcsaff** [comment]
100% coverage! That's a wrap.  It's all yours now, @lily-seabreeze

**Pull request closed without merging**